File: Patrones Comportamiento/Observer/Ejemplo.js
class Subject {
  constructor() { this.obs = []; }
  attach(o) { this.obs.push(o); }
  notify() { this.obs.forEach(o => o.update()); }
}

class Observer {
  update() { console.log("Notificado"); }
}

const s = new Subject();
s.attach(new Observer());
s.notify();

File: Patrones Comportamiento/Observer/Ejemplo.py
class Subject:
    def __init__(self):
        self.obs = []
    def attach(self, o):
        self.obs.append(o)
    def notify(self):
        for o in self.obs:
            o.update()

class Observer:
    def update(self):
        print("Notificado")

s = Subject()
s.attach(Observer())
s.notify()
